### ISSUE TITLE
 unexpected segment when trans a small graph

### ISSUE THREAD

**Issue by kiddyjinjin** [NVIDIA/TRTorch#454]
##  Bug Description

`terminate called after throwing an instance of 'pwgen::PwgenException'
  what():  Driver error:`

 I try to compile the torchscript graph  showed below using the latest trtorch.  and meet a core dump.

my graph struct:
 graph(%39 : Float(1, 2, 512, 64, strides=[65536, 64, 128, 1], requires_grad=0, device=cuda:0),
       %50 : Float(1, 1, 512, 512, strides=[262144, 262144, 512, 1], requires_grad=0, device=cuda:0),
       %56 : Float(1, 2, 512, 64, strides=[65536, 64, 128, 1], requires_grad=0, device=cuda:0),
       %59 : Float(1, 2, 512, 64, strides=[65536, 64, 128, 1], requires_grad=0, device=cuda:0)):
   %21 : int = prim::Constant[value=9223372036854775807]() 
   %29 : int[] = prim::Constant[value=[1, 512, 128]]()
   %32 : int = prim::Constant[value=0]() 
   %self.classifiers.11.self_atten.heads_num : int = prim::Constant[value=2]()
   %43 : bool = prim::Constant[value=0]()
   %42 : float = prim::Constant[value=0.5]() 
   %47 : None = prim::Constant() 
   %51 : int = prim::Constant[value=1]() 
   %54 : float = prim::Constant[value=8.]()
   %61 : int = prim::Constant[value=-1]() 
   %60 : int = prim::Constant[value=-2]() 
   %self.classifiers.11.self_atten.final_linear.weight : Float = prim::Constant[value=<Tensor>]()
   %self.classifiers.11.self_atten.final_linear.bias : Float = prim::Constant[value=<Tensor>]()
   %self.classifiers.11.output_layer_1.weight : Float = prim::Constant[value=<Tensor>]()
   %self.classifiers.11.output_layer_1.bias : Float = prim::Constant[value=<Tensor>]()
   %self.classifiers.11.output_layer_2.weight : Float = prim::Constant[value=<Tensor>]()
   %self.classifiers.11.output_layer_2.bias : Float = prim::Constant[value=0.01 *  4.6386 -4.8875 [ CUDAFloatType{2} ]]()
   %62 : Float = aten::transpose(%59, %60, %61) 
   %scores.38 : Float = aten::matmul(%56, %62) 
   %scores0.38 : Float = aten::div(%scores.38, %54) 
   %scores1.38 : Float = aten::add(%scores0.38, %50, %51) 
   %ret.49 : Float = aten::softmax(%scores1.38, %61, %47) 
   %probs0.38 : Float = aten::dropout(%ret.49, %42, %43) 
   %temp.38 : Float = aten::matmul(%probs0.38, %39)
   %37 : Float = aten::transpose(%temp.38, %51, %self.classifiers.11.self_atten.heads_num) 
   %33 : Float = aten::contiguous(%37, %32)  
   %output.62 : Float = aten::view(%33, %29)  
   %23 : Float = aten::slice(%hidden47.14, %32, %32, %21, %51)  
   %18 : Float = aten::select(%23, %51, %32)  
   %hidden53.14 : Float = aten::slice(%18, %51, %32, %21, %51)  
   %9 : Float = aten::linear(%hidden53.14, %self.classifiers.11.output_layer_1.weight, %self.classifiers.11.output_layer_1.bias)  
   %output_1.14 : Float = aten::tanh(%9)  
   %logits_this_layer10.1 : Float = aten::linear(%output_1.14, %self.classifiers.11.output_layer_2.weight, %self.classifiers.11.output_layer_2.bias) 
   return (%logits_this_layer10.1)

## Environment

TRTorch Version: 0.3.0
Using TensorRT Version: 7.2.3.4
PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.7.0 (Git Hash 7aed236906b1f7a05c0917e5257a1af05e9ff683)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 10.2
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70
  - CuDNN 7.6.5
  - Magma 2.5.2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=10.2, CUDNN_VERSION=7.6.5, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.8.1, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON,

## Additional context

Settings requested for TensorRT engine:
    Operating Precision: Float32
    TF32 Floating Point Computation Enabled: 1
    Truncate Long and Double: 0
    Make Refittable Engine: 0
    Debuggable Engine: 0
    Strict Types: 0
    GPU ID: 0
    Allow GPU Fallback (if running on DLA): 0
    Min Timing Iterations: 2
    Avg Timing Iterations: 1
    Max Workspace Size: 0
    Max Batch Size: Not set
    Device Type: GPU
    GPU ID: 0
    Engine Capability: Default
    Calibrator Created: 0

more debug info:

DEBUG: [TRTorch Conversion Context] - --------------- Timing Runner: <reformat> (Reformat)
DEBUG: [TRTorch Conversion Context] - Tactic: 1002 time 0.282752
DEBUG: [TRTorch Conversion Context] - Tactic: 0 time 0.087072
DEBUG: [TRTorch Conversion Context] - Fastest Tactic: 0 Time: 0.087072
DEBUG: [TRTorch Conversion Context] - --------------- Timing Runner: <reformat> (Reformat)
DEBUG: [TRTorch Conversion Context] - Tactic: 1002 time 0.0144
DEBUG: [TRTorch Conversion Context] - Tactic: 0 time 0.015264
DEBUG: [TRTorch Conversion Context] - Fastest Tactic: 1002 Time: 0.0144
DEBUG: [TRTorch Conversion Context] - --------------- Timing Runner: <reformat> (Reformat)
DEBUG: [TRTorch Conversion Context] - Tactic: 1002 time 0.281856
DEBUG: [TRTorch Conversion Context] - Tactic: 0 time 6.54336
DEBUG: [TRTorch Conversion Context] - Fastest Tactic: 1002 Time: 0.281856
DEBUG: [TRTorch Conversion Context] - *************** Autotuning format combination: Float(1,512,262144,524288), Float(1,512,262144,262144) -> Float(1,512,262144,524288) ***************
terminate called after throwing an instance of 'pwgen::PwgenException'
  what():  Driver error:
Aborted (core dumped)

**Comment by narendasan** [comment]
Im not sure what pwgen is. From a quick google search seems like the root cause of similar errors have been system configuration issues and not some sort of bug. 

Can you provide more context about your system configuration?

 - CPU Architecture:
 - OS (e.g., Linux):
 - How you installed PyTorch (`conda`, `pip`, `libtorch`, source):
 - Build command you used (if compiling from source):
 - Are you using local sources or building from archives:
 - Python version:
 - CUDA version:
 - GPU models and configuration:
 - Any other relevant information:

One think i noticed is you are using TensorRT 7.2.3.4 and PyTorch for CUDA 10.2. Did you modify the WORKSPACE to use a CUDA 10.2 based TensorRT? Our default target version is 11.1. I would say make sure your LD_LIBRARY_PATH does not include any stub libraries, that your CUDA versions across the various libraries are the same and that things like `nvidia-smi` dont report something like library/Driver version mismatch

**Comment by kiddyjinjin** [comment]
> Im not sure what pwgen is. From a quick google search seems like the root cause of similar errors have been system configuration issues and not some sort of bug.
> 
> Can you provide more context about your system configuration?
> 
> * CPU Architecture:
> * OS (e.g., Linux):
> * How you installed PyTorch (`conda`, `pip`, `libtorch`, source):
> * Build command you used (if compiling from source):
> * Are you using local sources or building from archives:
> * Python version:
> * CUDA version:
> * GPU models and configuration:
> * Any other relevant information:
> 
> One think i noticed is you are using TensorRT 7.2.3.4 and PyTorch for CUDA 10.2. Did you modify the WORKSPACE to use a CUDA 10.2 based TensorRT? Our default target version is 11.1. I would say make sure your LD_LIBRARY_PATH does not include any stub libraries, that your CUDA versions across the various libraries are the same and that things like `nvidia-smi` dont report something like library/Driver version mismatch

thanks. 
Here is my system configuration:
CPU Architecture:    x86_64   Intel(R) Xeon(R) Gold 6271C CPU @ 2.60GHz
OS (e.g., Linux):  linux   CentOS release 6.3 (Final)
How you installed PyTorch (conda, pip, libtorch, source):   compiled from source myself. 
Build command you used (if compiling from source):  ./bazelisk-linux-amd64 build //:libtrtorch --compilation_mode=dbg --distdir third_party/dist_dir/x86_64-linux-gnu  --cxxopt="-D_GLIBCXX_USE_CXX11_ABI=0" --linkopt="-D_GLIBCXX_USE_CXX11_ABI=0" --define=abi=pre_cxx11_abi --define=target_lang=python
Are you using local sources or building from archives:   local sources 
Python version:   python 3.6.5  with gcc8.2.0
CUDA version:   cuda-10.2
GPU models and configuration:  
Any other relevant information:

I have modified the WORKSPACE to adjust my system env 

- change  cuda-11.1  to  cuda-10.2  
- cudnn/secure/8.1.1.33/11.2_20210301   to  cudnn-8.0.5  
- TensorRT-7.2.3.4.CentOS-7.9.x86_64-gnu.cuda-10.2.cudnn8.1.tar.gz

and `nvidia-smi`  shows:
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 440.33.01    Driver Version: 440.33.01    CUDA Version: 10.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla T4            On   | 00000000:3B:00.0 Off |                    0 |
| N/A   28C    P8    10W /  70W |      0MiB / 15109MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   1  Tesla T4            On   | 00000000:5E:00.0 Off |                    0 |
| N/A   29C    P8     9W /  70W |      0MiB / 15109MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   2  Tesla T4            On   | 00000000:86:00.0 Off |                    0 |
| N/A   29C    P8     9W /  70W |      0MiB / 15109MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   3  Tesla T4            On   | 00000000:AF:00.0 Off |                    0 |
| N/A   28C    P8     9W /  70W |      0MiB / 15109MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+

**Comment by kiddyjinjin** [comment]
I  recompile the trtorch with 

- cuda 10.2  
- cudnn:  https://developer.nvidia.com/compute/machine-learning/cudnn/secure/8.1.1.33/10.2_20210301/cudnn-10.2-linux-x64-v8.1.1.33.tgz
-  libtorch:   https://download.pytorch.org/libtorch/cu102/libtorch-shared-with-deps-1.8.1%2Bcu102.zip
- tensorrt:  https://developer.nvidia.com/compute/machine-learning/tensorrt/secure/7.2.3/tars/TensorRT-7.2.3.4.CentOS-7.9.x86_64-gnu.cuda-10.2.cudnn8.1.tar.gz

and my problem solved.

thanks  for your  suggestion.  @narendasan
